content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx

```diff
@@ -9,40 +10,171 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
 
 </TopBlock>
 
-## <inlinecode> prisma migrate diff </inlinecode>
+### <inlinecode> migrate diff </inlinecode>
 
-This command examines two database schema sources to create a migration taking the first to the state of the second.
-The supported types of sources are:
+<Admonition type="info">
+
+This command is available in Preview in versions 3.9.0 and later.
+
+</Admonition>
+
+This command compares two database schema sources and outputs a description of a migration taking the first to the state of the second.
+
+The output can be given either as a human-readable summary (the default) or an executable script.
+
+The format of the command is:
+
+```terminal
+npx prisma migrate diff --preview-feature --from-... <source1> --to-... <source2>
+```
+
+where the `--from-...` and `--to-...` options are selected based on the type of database schema source. The supported types of sources are:
 
 - live databases
 - migration histories
 - Prisma data models
 - an empty schema
 
-The format of the command is:
+Both schema sources must use the same database provider. For example, a diff comparing a PostgreSQL data source with a SQLite data source is not supported.
 
-```terminal
-npx prisma migrate diff --preview-feature --from-empty --to-url $PROD_DATABASE
-```
+See also:
 
-For details of the available flags for this command, run:
+- [Migration troubleshooting in production](/guides/database/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute) <span class="guide"></span>
 
-```terminal
-npx prisma migrate diff --help
+#### Prerequisites
+
+Before using the `migrate diff` command, you must define a valid [`datasource`](/concepts/components/prisma-schema/data-sources) within your `schema.prisma` file.
```

Comment:
It's not a prerequisites as you can use from/to
- --from-url
- --from-migrations
- --from-schema-datamodel
- --from-empty`   

only --from-schema-datasource (or to) requires a valid datasource

Reply:
Thanks, done